backend/session 4/app.py

```diff
@@ -0,0 +1,334 @@
+            identity=email,
+            expires_delta=timedelta(weeks=52),
+        )
+        return jsonify(access_token=access_token), 200
+    return jsonify({"msg": "Invalid email or password"}), 401
+
+
+@app.route("/logout", methods=["POST"])
+@jwt_required()
+def logout():
+    return jsonify ("Successfully logged out"), 200
+
+@app.route("/delete_account", methods=["DELETE"])
+@jwt_required()
+def delete_account():
+    json=request.get_json()
+    current_user_email = get_jwt_identity()
+    user = User.query.filter_by(email=current_user_email).first()
+    password=json['password']
+    if user and bcrypt.check_password_hash(user.password_hash,password):
+        db.session.delete(user)
+        db.session.commit()
+        return jsonify("Account deleted"), 200
+    else:
+        return jsonify("Invalid"), 404
+
+
+@app.route("/profile", methods=["GET"])
+@jwt_required()
+def search():
+    current_user_email = get_jwt_identity()
+    user = User.query.filter_by(email=current_user_email).first()
+    if not (user is None):
+        return jsonify(
+            {
+                "username": user.username,
+                "age": user.age,
+                "gender": user.gender,
+                "school": user.school,
+                "email": user.email,
+            }
+        )
+    return jsonify("user not found"), 404
+
+
+@app.route("/edit_profile", methods=["PUT"])
+@jwt_required()
+def edit_profile():
+    json = request.get_json()
+    current_user_email = get_jwt_identity()
+    user = User.query.filter_by(email=current_user_email).first()
+    user.age = json["age"]
+    user.gender =json ["gender"]
+    user.school = json["school"]
+    db.session.commit()
+    return jsonify("updated successfully"), 200
+
+
+@app.route("/add_task", methods=["POST"])
+@jwt_required()
+def add():
+    json = request.get_json()
+    current_user_email = get_jwt_identity()
+    user = User.query.filter_by(email=current_user_email).first()
+    work = Task(your_work=json["text"], title=json["title"], user_id=user.id)
+    db.session.add(work)
+    db.session.commit()
+    return jsonify("received")
+
+
+@app.route("/remove_task", methods=["DELETE"])
+@jwt_required()
+def remove_task():
+    json = request.get_json()
+    current_user_email = get_jwt_identity()
+    user = User.query.filter_by(email=current_user_email).first()
+    task_remove = json["task_remove"]
+    task = Task.query.filter_by(your_work=task_remove, user_id=user.id).first()
+    if task:
+        db.session.delete(task)
+        db.session.commit()
+        return jsonify("Task Deleted")
+    return jsonify("Not Found"), 404
+
+
+@app.route("/edit_task", methods=["PUT"])
+@jwt_required()
+def edit_tasks():
+    json = request.get_json()
+    current_user_email = get_jwt_identity()
+    user = User.query.filter_by(email=current_user_email).first()
+    your_work = json["text"]
+    edit_task = Task.query.filter_by( user_id=user.id).first()
```

Comment:
Now this is wrong because you're quering on all user tasks
you should filter by your_work or the title

Reply:
use the same filter you did in "completed_tasks"

Reply:
Handle case where edit_task is None, return another status code (404 or bad request....etc)